From e2759e9fbd04162c47b849de06533b287194b234 Mon Sep 17 00:00:00 2001
From: Konstantin Makariev <kmakariev@motorola.com>
Date: Thu, 20 Aug 2015 15:55:13 -0500
Subject: [PATCH 034/959] IKKRNBSP-3337 arm/dt: 8952: adding touch node to
 devtree

Adding Synaptics touch device tree node.

Change-Id: I007e091dff2a07d839e51374ec6257ae6caea516
Signed-off-by: Konstantin Makariev <kmakariev@motorola.com>
Reviewed-on: http://gerrit.mot.com/778373
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi | 60 +++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index af7f29c1294..5c7c80d6240 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -25,6 +25,38 @@
 			qcom,pins = <&gp 85>;
 			qcom,num-grp-pins = <1>;
 	};
+
+	touch_reset_n {
+		qcom,pins = <&gp 64>;
+		qcom,num-grp-pins = <1>;
+		qcom,pin-func = <0>;
+		label = "touch-reset";
+
+		touch_reset_default: default {
+			output-low;
+		};
+
+		touch_reset_active: active {
+			drive-strength = <2>;
+			bias-pull-up; /* pull-up */
+		};
+	};
+
+	touch_int_n {
+		qcom,pins = <&gp 65>;
+		qcom,num-grp-pins = <1>;
+		qcom,pin-func = <0>;
+		label = "touch-int";
+
+		touch_int_default: default {
+			bias-disable;
+		};
+
+		touch_int_active: active {
+			drive-strength = <2>;
+			bias-pull-up; /* pull-up */
+		};
+	};
 };
 
 &soc {
@@ -145,3 +177,31 @@
 	qcom,platform-reset-gpio = <&msm_gpio 0 0>;
 	qcom,platform-te-gpio = <&msm_gpio 24 0>;
 };
+
+&i2c_3 { /* BLSP1 QUP3 */
+	synaptics@20 {
+		compatible = "synaptics,synaptics_dsx_i2c";
+		reg = <0x20>;
+		interrupt-parent = <&msm_gpio>;
+		interrupts = <65 0>;
+		gpios = <&msm_gpio 65 0>, /* IRQ */
+			<&msm_gpio 64 0>; /* RESET */
+		touch_vdd-supply = <&pm8950_l22>;
+		vdd_quirk-supply = <&pm8950_l6>;
+		/*vcc_i2c-supply = <&pm8950_l6>;*/
+		synaptics,gpio-config;
+		/* pins used by touchscreen */
+		pinctrl-names = "default", "active";
+		pinctrl-0 = <&touch_reset_default &touch_int_default>;
+		pinctrl-1 = <&touch_reset_active &touch_int_active>;
+		touchstate_modes {
+			touchmode-default = <&default_active &default_suspended>;
+		};
+		default_active: touchstate_default_active {
+			patch-data = "";
+		};
+		default_suspended: touchstate_aod_suspended {
+			patch-data = "F1@0=1&7";
+		};
+	};
+};
-- 
2.11.0

